### PR TITLE
Control Plane Listener proposal

### PR DESCRIPTION
**Pull request by scholzj**
This PR adds the Control Plane listener proposal, which builds on the feature gates proposal.

### CODE REVIEW

File: 025-control-plain-listener.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Strimzi currently has one internal listener.
+It is used by the Kafka brokers for data replication but also for coordination between the controller and the other brokers.
+And in addition to that, it is also used by the different Strimzi components:
+* Operators
+* Kafka Exporter
+* Cruise Control
+
+## Motivation
+
+Kafka already for some time supports using separate listeners for data replication and coordination.
```
Comment (tombentley) [inline 025-control-plain-listener.md:16]: Can we link to the KIP which added the feature?

**Review by tomncooper — APPROVED** [review]
This would be a great feature to have and should improve the resilience and performance of the Kafka Clusters. Using feature gates seems like a good strategy and one we will want to use more often in future so it will be good to try it out with this proposal.

File: 025-control-plain-listener.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+This can be easily handled while upgrading.
+But it is not possible to handle it when downgrading as the version the user would be downgrading from will not have a chance to revert the control plane listener changes.
+
+To mitigate this, this proposal suggests to use [Feature Gates](https://github.com/strimzi/proposals/blob/main/022-feature-gates.md) to introduce the control plane listener over multiple releases and cause minimal disruption for the users.
+Strimzi 0.23 will add new internal listener using port 9090.
+This listener will be enabled by default and configured in the same way as the replication listener.
+But it will not be configured as the control plane listener.
+
+A new feature gate `ControlPlaneListener` will be added and disabled by default.
+When enabled, this feature gate will configure the Kafka brokers to use the new listener for the control plane communication.
```
Comment (tombentley) [inline 025-control-plain-listener.md:35]: Does "the control plane communication" include the connections made by the CO to the brokers?

**Review by tombentley — APPROVED** [review]
A couple of minor comments, but this LGTM.

File: 025-control-plain-listener.md (hunk 1 above)
Comment (tomncooper) [reply 025-control-plain-listener.md:16]: Is it [KIP-103](https://cwiki.apache.org/confluence/display/KAFKA/KIP-103%3A+Separation+of+Internal+and+External+traffic) and/or [KIP-291](https://cwiki.apache.org/confluence/display/KAFKA/KIP-291%3A+Separating+controller+connections+and+requests+from+the+data+plane)?

Comment (tombentley) [reply 025-control-plain-listener.md:16]: It's 291

Comment (scholzj) [reply 025-control-plain-listener.md:16]: Yeah, its 291.

File: 025-control-plain-listener.md (hunk 2 above)
Comment (scholzj) [reply 025-control-plain-listener.md:35]: The plan expects them to keep using the replication listener. The Control Plane is only the Kafka control plane here. I updated the text to make it more clear.

**Review by sknot-rh — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**